### PR TITLE
Common PVC object

### PR DESCRIPTION
**Pull request by abays**
Wrapper object for PVCs

### CODE REVIEW

File: pkg/common/pvc.go [hunk 1]
```diff
@@ -19,50 +19,97 @@ package common
...
+	op, err := controllerutil.CreateOrPatch(ctx, h.GetClient(), pvc, func() error {
+		pvc.Annotations = MergeStringMaps(pvc.Annotations, p.pvc.Annotations)
+		pvc.Labels = MergeStringMaps(pvc.Labels, p.pvc.Labels)
 
-		pvc.Spec.Resources.Requests = corev1.ResourceList{
-			corev1.ResourceStorage: resource.MustParse(pv.Size),
-		}
+		pvc.Spec.Resources.Requests = p.pvc.Spec.Resources.Requests
+		pvc.Spec.StorageClassName = p.pvc.Spec.StorageClassName
+		pvc.Spec.AccessModes = p.pvc.Spec.AccessModes
```
[the remark this answers is not in the document]
Comment (abays) [reply pkg/common/pvc.go:62]: That's a good point that I have not checked.  Will do

[the remark this answers is not in the document]
Comment (abays) [reply pkg/common/pvc.go:62]: I'll just use the `CreationTimestamp` check for now and we can add more nuance later if needed

[the remark this answers is not in the document]
Comment (stuggi) [reply pkg/common/pvc.go]: could you add a comment in the function and commit message to mention the possible change of the size request in both and that for now we do not support that?

[the remark this answers is not in the document]
Comment (abays) [reply pkg/common/pvc.go]: Sure, will do

[the remark this answers is not in the document]
Comment (abays) [reply pkg/common/pvc.go]: Func/commit message updated

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**